### PR TITLE
Adding release process documentation to README.md

### PR DESCRIPTION
**Pull request by gvashishtha**
This PR better documents the release process for ML.NET

- [ x ] There's a descriptive title that will make sense to other developers some time from now. 
- [  ] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [ x ] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [ x ] You have included any necessary tests in the same PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +86,15 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
 |**Windows NetCore3.0**|[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/machinelearning/MachineLearning-CI?branchName=master&jobName=Windows_x64_NetCoreApp30&configuration=Windows_x64_NetCoreApp30%20Debug_Build)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=104&branchName=master)|[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/machinelearning/MachineLearning-CI?branchName=master&jobName=Windows_x64_NetCoreApp30&configuration=Windows_x64_NetCoreApp30%20Release_Build)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=104&branchName=master)|
 
+## Release process
+
+ML.NET has four kinds of releases: nightly builds, monthly preview releases, periodic general availability (GA) releases, and fix releases. We detail each kind of release below.
+
+1. **Nightly builds:** these are supported in NuGet, and support the most cutting-edge work. The trade-off is that these releases are not as rigorously tested.
```
Comment (CESARDELATORRE) [inline README.md]: @eerhardt @codemzs - I think we still don't have the Nightly builds as a public NuGet feed (at NuGet, MyGet or any public feed), right?

Afaik, the DevDiv feed at Azure DevOps is private
Can we have a public feed with the drops?

Comment (gvashishtha) [inline README.md]: @antoniovs1029 Is there a link I could link people to for this line? Not too familiar with NuGet, but I would want it to be a link that users could follow in order to get the nightly build.

Comment (antoniovs1029) [reply README.md]: Hi. Currently there's no public feed for the users to get NuGets from the nightly builds. I am working on this with @safern and hopefully we will be able to provide you with a link for this by the end of the week.

**Comment by codemzs** [comment]
@gvashishtha This PR has been outstanding for a long time. If there are no imminent plans to check this in then I suggest we close this and revisit when it is ready to merged.

File: README.md (hunk 1 above)
Comment (codemzs) [reply README.md]: We do now https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning

Comment (codemzs) [reply README.md]: https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning

**Comment by gvashishtha** [comment]
@codemzs please take a look at this, make sure it is consistent with our conversation today.

File: README.md [hunk 2]
```diff
@@ -86,6 +86,35 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
 
+## Release process and versioning
+
+ML.NET NuGets (of which there are approximately 25) are versioned with the following format: `A.B.C<-D>`, where `A`, `B`, and `C` are integers, and `D` is an optional string.
+
+- `A` - **version number**: If `A` is 0, this NuGet is considered a **work in progress (WIP)**, and could be deleted at any time. If `A` is greater than 0, then we plan to support the corresponding NuGet indefinitely. 
+- `B` - **sub-version number**: This number is consistent within each GA release and within each WIP release. Therefore, all GA NuGets that are released at the same time will have the same sub-version number, and all WIP releases that are released at the same time will have the same sub-version number.
+- `C` - **bug fix index**: `C` starts at 0 and is incremented every time we introduce a bug fix between releases.
```
Comment (codemzs) [inline README.md]: >bug fix index [](start = 10, length = 13)

this is called "patch" release number/index

Comment (codemzs) [inline README.md]: >bug fix [](start = 88, length = 7)

patch

File: README.md [hunk 3]
```diff
@@ -86,6 +86,35 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
+
+ML.NET has four kinds of releases: nightly builds, monthly preview releases, periodic general availability (GA) releases, and fix releases. We detail each kind of release below.
+
+1. **Nightly builds:** these can be downloaded from [this NuGet feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning), and are built automatically once per day, based on the content of the `master` branch.
+1. **Monthly preview releases:** These releases are built from the corresponding `A.B-preview-X` branch, and are expected to meet a higher quality bar than the nightly builds. These can also be downloaded from [this NuGet feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning), or within Visual Studio, as detailed below.
+1. **GA releases:** These releases are built from the corresponding `A.B` branch. They are rigorously tested, stable, and meant for general use. They are also the default choice when installing ML.NET via the `Install-Package Microsoft.ML` command.
+1. **Fix releases:** These releases include patches for bugs in either the preview or GA releases.
```
Comment (codemzs) [inline README.md]: >Fix [](start = 5, length = 3)

Patch

File: README.md [hunk 4]
```diff
@@ -86,6 +86,35 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
+- `B` - **sub-version number**: This number is consistent within each GA release and within each WIP release. Therefore, all GA NuGets that are released at the same time will have the same sub-version number, and all WIP releases that are released at the same time will have the same sub-version number.
+- `C` - **bug fix index**: `C` starts at 0 and is incremented every time we introduce a bug fix between releases.
+- `D` - **preview suffix**: `D` is an optional suffix which contains the word "preview" followed by an integer. If D is not included and A is not 0, then the API surface is locked.
+
+ML.NET has four kinds of releases: nightly builds, monthly preview releases, periodic general availability (GA) releases, and fix releases. We detail each kind of release below.
+
+1. **Nightly builds:** these can be downloaded from [this NuGet feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning), and are built automatically once per day, based on the content of the `master` branch.
```
Comment (codemzs) [inline README.md]: >and are built automatically once per day [](start = 148, length = 40)

Incorrect, they are built when a commit is made to the master branch.

Comment (codemzs) [inline README.md]: >Nightly [](start = 5, length = 7)

Daily builds

File: README.md [hunk 5]
```diff
@@ -86,6 +86,35 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
+1. **Monthly preview releases:** These releases are built from the corresponding `A.B-preview-X` branch, and are expected to meet a higher quality bar than the nightly builds. These can also be downloaded from [this NuGet feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning), or within Visual Studio, as detailed below.
```
Comment (codemzs) [inline README.md]: >releases [](start = 21, length = 8)

Not sure if we need to mention "releases" each time the heading says below are the different kinds of releases ...

File: README.md [hunk 6]
```diff
@@ -86,6 +86,35 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
+1. **GA releases:** These releases are built from the corresponding `A.B` branch. They are rigorously tested, stable, and meant for general use. They are also the default choice when installing ML.NET via the `Install-Package Microsoft.ML` command.
+1. **Fix releases:** These releases include patches for bugs in either the preview or GA releases.
+
+The table below explains how each of the elements in our versioning schema would change for each kind of release.
+`
+| Release type | Change in `A` | Change in `B` | Change in `C` | Change in `D` |
+| -------------|-------------|-------------|-------------|-------------|
+| Nightly build| No change   | No change   | No change   | No change   |
```
Comment (codemzs) [inline README.md]: > Nightly build| No change   | No change   | No change   | No change   | [](start = 1, length = 71)

Daily builds have a date like stamp at the end example Microsoft.Extensions.ML 1.5.0-preview-28327-2

File: README.md [hunk 7]
```diff
@@ -86,6 +86,35 @@ To build ML.NET from source please visit our [developers guide](docs/project-doc
...
+| Monthly preview| No change | No change   | No change   | `preview` tag added, if this is the first preview, or preview index incremented (i.e. `A.B.C-preview` -> `A.B.C-preview2`) |
+| GA releases | Possibly incremented for non-WIP NuGets | Incremented | Reset to 0 | `preview` tag is removed |
```
Comment (codemzs) [inline README.md]: monthly releases can also be GA releases, like 1.4.0

Comment (gvashishtha) [reply README.md]: OK, so in months when there is a monthly GA release, there is no monthly preview release? Like, when 1.4.0 came out, there was no 1.4-preview?

Comment (harishsk) [reply README.md]: The idea of preview releases is for our early adopters to preview new apis and features and give us feedback. So, ideally our GA releases should not include new apis and features and should mostly be bug fixes, doc updates, perf improvements and things like that. When we do that, there is no need for a preview release when we make a GA release.

Comment (gvashishtha) [reply README.md]: @harishsk do let me know if the new version is good to go.

File: docs/project-docs/release-process.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+1. **Daily builds:** these can be downloaded from [this NuGet feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning), and are built automatically each time a commit is made to the `master` branch.
+1. **Preview:** These releases are built from the corresponding `A.B-preview-X` GitHub branch, and are expected to meet a higher quality bar than the daily builds. These can also be downloaded from [this NuGet feed](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=MachineLearning), or within Visual Studio, as detailed below. When we introduce new API's in a preview release, we avoid doing a GA release at the same time (unless there are patches required for the last GA release). If there are no new API's, then we go straight to a GA release and skip the preview release.
+1. **GA:** These releases are built from the corresponding `A.B` GitHub branch. They are rigorously tested, stable, and meant for general use. They are also the default choice when installing ML.NET via the `Install-Package Microsoft.ML` command.
+1. **Fix:** These releases include patches for bugs in either the preview or GA releases.
```
Comment (harishsk) [inline docs/project-docs/release-process.md]: Please add that the GA releases are published to nuget.org and this should be good to go.

**Comment by najeeb-kazmi** [comment]
@gvashishtha can you please pull master so that the checks pass?

**Pull request merged**